src/tools/copyset_check_core.cpp

```diff
@@ -148,12 +148,16 @@ int CopysetCheckCore::CheckCopysetsOnChunkServer(
 }
 
 ChunkServerHealthStatus CopysetCheckCore::CheckCopysetsOnChunkServer(
-                    const std::string& chunkserverAddr,
-                    const std::set<std::string>& groupIds,
-                    bool queryLeader) {
+                            const std::string& chunkserverAddr,
+                            const std::set<std::string>& groupIds,
+                            bool queryLeader, butil::IOBuf *iobuf, int res) {
     bool isHealthy = true;
-    butil::IOBuf iobuf;
-    int res = QueryChunkServer(chunkserverAddr, &iobuf);
+    butil::IOBuf tmpBuf;
+    if (nullptr == iobuf) {
```

Comment:
event it don't have some problems, but it's just looks wired.
I think is better to encapsulate last two parameters into a struct/class, and add another state which indicates whether we should `QueryChunkServer` or not.

Reply:
> event it don't have some problems, but it's just looks wired.
> I think is better to encapsulate last two parameters into a struct/class, and add another state which indicates whether we should `QueryChunkServer` or not.

fixed